docs/pages/api/telemetry/index.ts

```diff
@@ -0,0 +1,38 @@
+import { Analytics } from "@segment/analytics-node";
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default async function forwardTelemetry(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method !== "POST") {
+    res.setHeader("Allow", ["POST"]);
+    return res.status(405).end(`Method ${req.method} Not Allowed`);
+  }
+
+  if (!process.env.SEGMENT_WRITE_KEY) {
+    console.error('Missing required environment variable "SEGMENT_WRITE_KEY".');
+    return res
+      .status(500)
+      .json({ error: "An error occurred while processing telemetry data." });
+  }
+
+  try {
+    const analytics = new Analytics({
+      writeKey: process.env.SEGMENT_WRITE_KEY,
+    });
+
+    await analytics.track({
+      ...req.body,
+    });
+
+    res.status(200).json({
+      message: "Telemetry data processed successfully.",
+    });
```

Comment:
Do we need to await the request to Segment? I wonder if we can actually send the HTTP response back to the client immediately, then try to forward to segment.

Reply:
we can do that - the tradeoff is that the client will never know if the event was forwarded successfully. In this case, I don't think it is important but it'd be good to track if the forwarding fails

Reply:
Yea. I wonder if Vercel has a dashboard for "failed edge function invocations" or something. Like, maybe we can just throw in the function body if the `analytics.track` call fails. In any case, this works as is for now.

I don't think the client should know/care if the forwarding failed - there's nothing they can do about it.